applications/dashboard/models/class.userauthenticationnoncemodel.php

```diff
@@ -1,25 +1,35 @@
 <?php
 /**
- * UserAuthenticationNonce model.
- *
+ * @author Chris Chabilall chris.c@vanillaforums.com
  * @copyright 2009-2018 Vanilla Forums Inc.
  * @license http://www.opensource.org/licenses/gpl-2.0.php GNU GPL v2
  */
 
 /**
- * Handles user data.
+ * Handles user data and issuing and consuming nonces.
  */
 class UserAuthenticationNonceModel extends Gdn_Model {
 
     use \Vanilla\PrunableTrait;
+    use \Vanilla\TokenSigningTrait;
+
+    /**
+     * The timestamp of a consumed nonce.
+     */
+    const CONSUMED_TIMESTAMP = "1971-01-01 00:00:01";
 
     /**
      * Class constructor. Defines the related database table name.
+     *
+     * @param string $secret The secret used to sign access tokens for the client.
      */
-    public function __construct() {
+    public function __construct($secret = null) {
         parent::__construct('UserAuthenticationNonce');
         $this->setPruneField('Timestamp');
         $this->setPruneAfter('45 minutes');
+        $this->setSecret($secret);
```

Comment:
Bringing back what I said before: Can we have this default to the value of `Garden.Cookie.Salt`?

I'm not seeing any current usage of `UserAuthenticationNonceModel` passing anything to the constructor. That would mean after this change, there wouldn't be a secret to sign any of the nonces.

Reply:
I was thinking of something like this

    public function setSecret($secret) {
        $this->secret = $secret ?: c('Garden.Cookie.Salt');
        return $this;
    }

Reply:
I would stick to letting the class determine how a default should be set. `AccessTokenModel` and `UserAuthenticationNonce` could maybe default to that config, but maybe another class implementing the trait would have a separate source for its default secret.